docs/api-markdown-documenter/index.js

```diff
@@ -3,17 +3,38 @@
  * Licensed under the MIT License.
  */
 
-const chalk = require("chalk");
+/**
+ * This index script runs the renderApiDocumentation script using the version configurations described 
+ * in data/versions.json. This script allows for an optional boolean parameter which determines whether 
+ * renderApiDocumentation will be ran for all versions or only previous versions. (pass in true for all versions)
+ * e.g. "node ./api-markdown-documenter/index.js true"
+ */
 
+const chalk = require("chalk");
+const versions = require("../data/versions.json");
 const { renderApiDocumentation } = require("./render-api-documentation");
 
-renderApiDocumentation().then(
-	() => {
-		console.log(chalk.green("API docs written!"));
-		process.exit(0);
-	},
-	(error) => {
-		console.error("API docs could not be written due to an error:", error);
-		process.exit(1);
-	},
-);
+const renderMultiVersion = process.argv[2];
+
+docVersions = renderMultiVersion
+	? versions.params.previousVersions.concat(versions.params.currentVersion)
+	: [versions.params.currentVersion];
+
+const apiDocRenders = [];
+
+docVersions.forEach((version) => {
+	apiDocRenders.push(
+		renderApiDocumentation(version).then(
+			() => {
+				console.log(chalk.green("API docs written!"));
+				process.exit(0);
+			},
+			(error) => {
+				console.error("API docs could not be written due to an error:", error);
+				process.exit(1);
+			},
+		),
+	);
+});
+
+Promise.all(apiDocRenders);
```

Comment:
Tong and I were looking at this on her PR, and it occurred to me that we should change the error handling here a bit. As it is, if 1 of our render passes fails, we exit the process immediately (hiding any other potential errors we could have found in other render passes). I would recommend something like the following:
```suggestion
docVersions.forEach((version) => {
	apiDocRenders.push(
		renderApiDocumentation(version).then(
			() => {
				console.log(chalk.green(`${version} API docs written!`));
			},
			(error) => {
				throw new error(`${version} API docs could not be written due to an error:`, error);
			},
		),
	);
});

Promise.all(apiDocRenders).then(
	() => {
		console.log(chalk.green("All API docs written!"));
		process.exit(0);
	},
	(error) => {
		console.error(error);
		process.exit(1);
	},
);
```